src/lib/browser.ts

```diff
@@ -125,9 +129,13 @@ export class Browser extends EventEmitter {
     }
     
     private addService(service: Service) {
-        this._services.push(service)
-        this.serviceMap[service.fqdn] = true
-        this.emit('up', service)
+        try {
```

Comment:
try catch block is probably not needed here; it is better not to ignore an exception.